### PR TITLE
feat: add getShareableUrl method to navigation service

### PR DESCRIPTION
**Pull request by arjunlalb**
## Description
Expose a method to construct a shareableUrl that contains the current domain info.

### Testing
Added UT

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/common/src/navigation/navigation.service.ts [hunk 1]
```diff
@@ -272,8 +272,8 @@ export class NavigationService {
     return this.platformLocation.href;
   }
 
-  public getHostUrl(): string {
-    return new URL(this.getAbsoluteCurrentUrl()).origin;
+  public getShareableUrl(path: string[], queryParams?: Dictionary<string>): string {
```
Comment (aaron-steinfeld) [inline projects/common/src/navigation/navigation.service.ts:275]: Could we make this interface match some of the other ones? They're a little inconsistent but either taking NavParams like our main nav method
```
navigate(paramsOrUrl: NavigationParams | string)
```
or at least taking the query params as a Params object like getUrlTree (not sure what the type of UrlSegment is - if it's an alias for string, even better - if it needs to be constructed, string[] is fine)
```
getUrlTree(urlSegments: UrlSegment[], queryParams?: Params)
```

Comment (arjunlalb) [reply projects/common/src/navigation/navigation.service.ts]: done.

File: projects/common/src/navigation/navigation.service.test.ts [hunk 2]
```diff
@@ -50,7 +50,8 @@ describe('Navigation Service', () => {
     providers: [
       mockProvider(Location),
       mockProvider(Title, { setTitle: jest.fn().mockReturnValue(undefined) }),
-      { provide: APP_TITLE, useValue: 'defaultAppTitle' }
+      { provide: APP_TITLE, useValue: 'defaultAppTitle' },
+      mockProvider(PlatformLocation, { href: 'https://test.traceable.ai' })
```
Comment (aaron-steinfeld) [inline projects/common/src/navigation/navigation.service.test.ts:54]:
```suggestion
      mockProvider(PlatformLocation, { href: 'https://test.hypertrace.orgi' })
```

Comment (arjunlalb) [reply projects/common/src/navigation/navigation.service.test.ts]: done.

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**